tests/test_ir.py

```diff
@@ -20,3 +22,73 @@ def test_ops_accessor():
     d = Subi.get(a, b)
 
     assert d.results[0] != c.results[0]
+
+
+def test_ops_accessor_II():
+    a = Constant.from_int_and_width(1, i32)
+    b = Constant.from_int_and_width(2, i32)
+    # Operation to add these constants
+    c = Addi.get(a, b)
+
+    block0 = Block.from_ops([a, b, c])
+    # Create a region to include a, b, c
+    region = Region.from_block_list([block0])
+
+    assert len(region.ops) == 3
+    assert len(region.blocks[0].ops) == 3
+
+    # Operation to subtract b from a
+    d = Subi.get(a, b)
+
+    assert d.results[0] != c.results[0]
+
+    # Erase operations and block
+    region2 = Region()
+    region.move_blocks(region2)
+
+    region2.blocks[0].erase_op(a, safe_erase=False)
+    region2.blocks[0].erase_op(b, safe_erase=False)
+    region2.blocks[0].erase_op(c, safe_erase=False)
+
+    region2.detach_block(block0)
+    region2.drop_all_references()
+
+    assert len(region2.blocks) == 0
+
+
+def test_ops_accessor_III():
+    # Create constants `from_attr` and add them, add them in blocks, blocks in
+    # a region and create a function
+    a = Constant.from_attr(IntegerAttr.from_int_and_width(1, 32), i32)
+    b = Constant.from_attr(IntegerAttr.from_int_and_width(2, 32), i32)
+    c = Constant.from_attr(IntegerAttr.from_int_and_width(3, 32), i32)
+    d = Constant.from_attr(IntegerAttr.from_int_and_width(4, 32), i32)
+
+    # Operation to add these constants
+    e = Addi.get(a, b)
+    f = Addi.get(c, d)
+
+    # Create Blocks and Regions
+    block0 = Block.from_ops([a, b, e])
+    block1 = Block.from_ops([c, d, f])
```

Comment:
I feel like only having the two constants here and no Add would make the block simpler while still testing the same thing.

Also, I am continuing to feel like we should have an `testing.test_op` that is basically just a substitute for a generic operation and can be used for such tests, where it doesn't really matter that this is a Constant, but it should just be some operation. This would remove the dependency on the Arith dialect for these tests. What do you think?

Reply:
I am not sure how exactly do you think of this implementation. Could you elaborate?

Reply:
I think this should just be its own dialect, the `testing` dialect. I feel that unit tests should be as independent as possible and creating dependencies among tests (like here, the IR tests depend on the Arith dialect) is not a smart way to go about testing in general. Though this is a more general point and not specific to this aprticular PR :)

Reply:
I agree, we could do this in the workshop and another PR ;)